scripts/prepare-test-cluster.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+set -x
+# Setup to find nessasary data from cluster setup
+# Constants
+HTPASSWD_FILE="./htpass"
+USERPASS="developer"
+HTPASSWD_SECRET="htpasswd-secret"
+SETUP_OPERATORS="./scripts/setup-operators.sh"
+# Overrideable information
+DEFAULT_INSTALLER_ASSETS_DIR=${DEFAULT_INSTALLER_ASSETS_DIR:-$(pwd)}
+KUBEADMIN_USER=${KUBEADMIN_USER:-"kubeadmin"}
+KUBEADMIN_PASSWORD_FILE=${KUBEADMIN_PASSWORD_FILE:-"${DEFAULT_INSTALLER_ASSETS_DIR}/auth/kubeadmin-password"}
+# Default values
+OC_STABLE_LOGIN="false"
+# Exported to current env
+ORIGINAL_KUBECONFIG=${KUBECONFIG:-"${DEFAULT_INSTALLER_ASSETS_DIR}/auth/kubeconfig"}
+export KUBECONFIG=$ORIGINAL_KUBECONFIG
+
+# list of namespace to create
+OPERATOR_NAMESPACES="cicd argocd"
+
+# Attempt resolution of kubeadmin, only if a CI is not set
+if [ -z $CI ]; then
+    # Check if nessasary files exist
+    if [ ! -f $KUBEADMIN_PASSWORD_FILE ]; then
+        echo "Could not find kubeadmin password file"
+        exit 1
+    fi
+
+    if [ ! -f $KUBECONFIG ]; then
+        echo "Could not find kubeconfig file"
+        exit 1
+    fi
+
+    # Get kubeadmin password from file
+    KUBEADMIN_PASSWORD=`cat $KUBEADMIN_PASSWORD_FILE`
+
+    # Login as admin user
+    oc login -u $KUBEADMIN_USER -p $KUBEADMIN_PASSWORD
+else
+    # Copy kubeconfig to temporary kubeconfig file
+    # Read and Write permission to temporary kubeconfig file
+    TMP_DIR=$(mktemp -d)
+    cp $KUBECONFIG $TMP_DIR/kubeconfig
+    chmod 640 $TMP_DIR/kubeconfig
+    export KUBECONFIG=$TMP_DIR/kubeconfig
+fi
+
+# Create the namespace for operator installation namespace
+for i in `echo $OPERATOR_NAMESPACES`; do
+    # create the namespace
+    oc new-project $i
+    # Let developer user have access to the project
+    oc adm policy add-role-to-user edit developer
+done
+
+# Setup the cluster for sealed secrets, pipelines and argocd operator
+sh $SETUP_OPERATORS
+
+# Remove existing htpasswd file, if any
+if [ -f $HTPASSWD_FILE ]; then
+    rm -rf $HTPASSWD_FILE
+fi
+
+# Set so first time -c parameter gets applied to htpasswd
+HTPASSWD_CREATED=" -c "
+
+# Create htpasswd entries for developer
+htpasswd -b $HTPASSWD_CREATED $HTPASSWD_FILE developer $USERPASS
+HTPASSWD_CREATED=""
+
+# Create secret in cluster, removing if it already exists
```

Comment:
Is there a reason not to use `oc replace` instead of delete/create?

Reply:
I am generating secret through ```htpasswd``` command and storing it in a regular file. oc replace does not provide such option to get password from a file.

```
$ oc replace secret -h | grep "from-file"
$
```

Reply:
I think something like this should work :-)

```shell
$ oc create secret generic ${HTPASSWD_SECRET} --from-file=htpasswd=${HTPASSWD_FILE} \
  -n openshift-config --dry-run=client -o yaml | oc apply -f -

Reply:
+1, Added

Reply:
```oc replace -f -``` does not work on a fresh cluster as it does not find the existing secret. So i used apply instead of replace

Reply:
Doh, I had apply in there from a script I use, and changed it to replace without really thinking :-)